include/linker/devicetree_regions.h

```diff
@@ -24,3 +24,12 @@
 	COND_CODE_1(DT_NODE_HAS_STATUS(node, okay),	  \
 		    (_REGION_DECLARE(name, attr, node)),  \
 		    ())
+
+/**
+ * @brief Get a memory region name from a devicetree node
+ *
+ * @param node_id node identifier
+ * @retval unquoted node label
+ */
+#define DT_REGION_NAME(node_id)	\
+	DT_NODE_NODELABEL(node_id, UTIL_DECR(DT_NODE_NUM_NODELABEL(node_id)))
```

Comment:
This doesn't seem like a good idea to me. It seems quite fragile. Users should be free to add whatever kind of node labels they want to any nodes without modifying things like this.

Reply:
As long as users use `DT_REGION_NAME(node_id)` in linker scripts, it is not possible to break this setup AFAICT.
Any changes in the name caused by overlays will automatically propagate through to the linker scripts.
Users are absolutely free to add labels as they wish, its just that the last one applied will be the name generated by the linker.